src/player.cpp

```diff
@@ -4168,21 +4168,21 @@ int player::overmap_sight_range(int light_level) const
     if( sight <= SEEX * 4) {
         return (sight / (SEEX / 2) );
     }
-    if ( has_item_with_flag( "ZOOM" ) &&
```

Comment:
Good idea. But is this line right? Did you mean "or"?
} else if( has_optic != has_trait( "EAGLEEYED" ) ) {